WMF Framework/WMFExploreFeedContentController.m

```diff
@@ -386,31 +406,41 @@ -(void)toggleContentForSiteURL:(NSURL *)siteURL isOn:(BOOL)isOn updateFeed:(BOOL
 }
 
 - (void)toggleContentGroupOfKind:(WMFContentGroupKind)contentGroupKind forSiteURLs:(NSSet<NSURL *> *)siteURLs isOn:(BOOL)isOn {
+    [self toggleContentGroupKinds:[NSSet setWithObject:@(contentGroupKind)] forSiteURLs:siteURLs isOn:isOn];
+}
+
+- (void)toggleContentGroupKinds:(NSSet <NSNumber *> *)contentGroupKindNumbers forSiteURLs:(NSSet<NSURL *> *)siteURLs isOn:(BOOL)isOn {
     [self updateExploreFeedPreferences:^(NSMutableDictionary *newPreferences) {
         for (NSURL *siteURL in siteURLs) {
-            NSString *key = siteURL.wmf_articleDatabaseKey;
-            NSSet *oldVisibleContentSources = [newPreferences objectForKey:key];
-            NSMutableSet *newVisibleContentSources;
-
-            if (oldVisibleContentSources) {
-                newVisibleContentSources = [oldVisibleContentSources mutableCopy];
-            } else {
-                newVisibleContentSources = [NSMutableSet set];
-            }
+            for (NSNumber *contentGroupKindNumber in contentGroupKindNumbers) {
+                NSString *key = siteURL.wmf_articleDatabaseKey;
+                NSSet *oldVisibleContentGroupKindNumbers = [newPreferences objectForKey:key];
+                NSMutableSet *newVisibleContentGroupKindNumbers;
 
-            if (isOn) {
-                [newVisibleContentSources addObject:@(contentGroupKind)];
-            } else {
-                [newVisibleContentSources removeObject:@(contentGroupKind)];
-            }
+                if (oldVisibleContentGroupKindNumbers) {
+                    newVisibleContentGroupKindNumbers = [oldVisibleContentGroupKindNumbers mutableCopy];
+                } else {
+                    newVisibleContentGroupKindNumbers = [NSMutableSet set];
+                }
+
+                if (isOn) {
+                    [newVisibleContentGroupKindNumbers addObject:contentGroupKindNumber];
+                } else {
+                    [newVisibleContentGroupKindNumbers removeObject:contentGroupKindNumber];
+                }
 
-            [newPreferences setObject:newVisibleContentSources forKey:key];
+                [newPreferences setObject:newVisibleContentGroupKindNumbers forKey:key];
+            }
         }
     } completion:^{
         [self updateFeedSourcesUserInitiated:YES completion:nil];
     }];
 }
 
+- (void)toggleGlobalContentGroupKinds:(BOOL)on {
+    [self toggleContentGroupKinds:[WMFExploreFeedContentController globalContentGroupKindNumbers] forSiteURLs:self.preferredSiteURLs isOn:on];
```

Comment:
Does this stay in sync if the user changes their preferred site URLs? Could this be stored as a boolean instead of having to be updated for every site URL?

Reply:
Would need to alter `applyExploreFeedPreferencesToObjects` to first check if it's a global content group kind, if so, check the boolean, otherwise do the normal siteURL check